### DISCUSSION TITLE
where is the 'isTokenExpired' function in 'pocketbase/cjs'?

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#4449] [Q&A]
I'm migrating my code to TS, and after searching the sdk-js repository, I discovered that I need to import 'pocketbase/cjs' for it to work.
everything's fine so far.
however, when trying to use the 'isTokenExpired' function, an error appears stating that it does not exist.
How to import it correctly?
below my code:

```
//@ts-ignore
import PocketBase, { isTokenExpired } from 'pocketbase/cjs';

const host = 'http://example.com'
const pb = new PocketBase(host);
pb.autoCancellation(false);

export default async function initPB() {
    if (pb.authStore.isValid) {
        if (isTokenExpired(pb.authStore.token, 1800)) {
            await pb.admins.authRefresh()
        }
    } else {
        // authenticate with email/password
        await pb.admins.authWithPassword('example@example.com', 'password123');
    }
    return pb
}
```

When trying to run this error appears in the console:

```
TypeError: (0 , cjs_1.isTokenExpired) is not a function
    at initPB (D:\codigos\WPPBotAPI\pocketbaseAdmin.js:33:38)
    at saveKey (D:\codigos\WPPBotAPI\usePocketBaseAuthState.js:20:50)
    at writeData (D:\codigos\WPPBotAPI\usePocketBaseAuthState.js:97:15)
    at usePocketBaseAuthState (D:\codigos\WPPBotAPI\usePocketBaseAuthState.js:135:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async createSession (D:\codigos\WPPBotAPI\wpp.js:61:36)
```

I tried countless ways to import this function, but without success.
How to use it in typescript correctly?

**Comment by ganigeorgiev** [comment]
Non-default exports are not accessible when using the CJS bundle.

>  I discovered that I need to import 'pocketbase/cjs' for it to work.

No, it shouldn't be needed. The CJS bundle should be considered legacy and it is there only for backwards-compatibility. Double check your TS configuration and use the ESM bundle.

**Reply by erickythierry** [reply]
Thanks for the answer.
Could you give me an example of how it should be? I'm learning how TS works, so I'm a little lost with these errors.

**Reply by ganigeorgiev** [reply]
>  I'm learning how TS works, so I'm a little lost with these errors.

Could you post the error that it is returned when just using the regular `from 'pocketbase'` import?

It would also help if you post your current TS config.

**Reply by erickythierry** [reply]
Thanks for your help, I managed to solve it after researching a little about what you told me in your first comment.
now it worked normally. However, I don't know if my tsconfig.json is using a current version to generate the code.

```
{
  "compilerOptions": {
    "target": "ES6",
    "module": "ES6",
    "moduleResolution": "node",
    "outDir": "./",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  },
  "include": [
    "./src/**/*.ts",
  ],
  "exclude": [
    "node_modules"
  ]
}
```

Could you tell me if it's ok?
